File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s)
{

    // 모두 소문자 변환
    for(auto& c : s)
        if(isupper(c))
            c = tolower(c);

    // 개수 새기
    int p_cnt = 0;
    int y_cnt = 0;
    
    for(auto& c : s){
        if(c == 'p')
            p_cnt++;
        if(c == 'y')
            y_cnt++;
    }
    
    return p_cnt == y_cnt;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        
        s = s.toLowerCase();
        
        int p_cnt = 0;
        int y_cnt = 0;
        
        for(char c : s.toCharArray()){
            if(c == 'p')
                p_cnt++;
            else if(c == 'y')
                y_cnt++;
        }
        
        return p_cnt == y_cnt;
    }
}